### PR TITLE
[cranelift docs] Fix grammar code blocks and heap image

### PR DESCRIPTION
**Pull request by samrat**
- Move grammar segments in IR docs to code blocks
- Replace dot code in IR docs with generated image.

### CODE REVIEW

File: cranelift/docs/ir.md [hunk 1]
```diff
@@ -95,11 +95,13 @@ execution can never fall through to the next EBB without an explicit branch.
 
 A `.clif` file consists of a sequence of independent function definitions:
 
-.. productionlist::
-    function_list : { function }
-    function      : "function" function_name signature "{" preamble function_body "}"
-    preamble      : { preamble_decl }
-    function_body : { extended_basic_block }
+```
+productionlist::
```
Comment (bjorn3) [inline cranelift/docs/ir.md]:
```suggestion
```

That was a `rst` annotation for the code block language, just like

````
```productionlist
````

for markdown.

File: cranelift/docs/ir.md [hunk 2]
```diff
@@ -397,14 +399,16 @@ be known when the function call is compiled. The function signature describes
...
-.. productionlist::
-    signature    : "(" [paramlist] ")" ["->" retlist] [call_conv]
-    paramlist    : param { "," param }
-    retlist      : paramlist
-    param        : type [paramext] [paramspecial]
-    paramext     : "uext" | "sext"
-    paramspecial : "sret" | "link" | "fp" | "csr" | "vmctx" | "sigid" | "stack_limit"
-    callconv     : "fast" | "cold" | "system_v" | "fastcall" | "baldrdash_system_v" | "baldrdash_windows"
+```
+productionlist::
```
Comment (bjorn3) [inline cranelift/docs/ir.md]:
```suggestion
```

File: cranelift/docs/heap.svg [hunk 3]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN"
+ "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
+<!-- Generated by graphviz version 2.42.3 (0)
```
Comment (bjorn3) [inline cranelift/docs/heap.svg]: Maybe keep the original `.dot` file somewhere?

File: cranelift/docs/ir.md [hunk 4]
```diff
@@ -95,11 +95,12 @@ execution can never fall through to the next EBB without an explicit branch.
...
-.. productionlist::
-    function_list : { function }
-    function      : "function" function_name signature "{" preamble function_body "}"
-    preamble      : { preamble_decl }
-    function_body : { extended_basic_block }
+```
+ function_list : { function }
+ function      : "function" function_name signature "{" preamble function_body "}"
+ preamble      : { preamble_decl }
+ function_body : { extended_basic_block }
```
Comment (bjorn3) [inline cranelift/docs/ir.md]:
```suggestion
function_list : { function }
function      : "function" function_name signature "{" preamble function_body "}"
preamble      : { preamble_decl }
function_body : { extended_basic_block }
```

File: cranelift/docs/ir.md [hunk 5]
```diff
@@ -397,14 +398,15 @@ be known when the function call is compiled. The function signature describes
...
-    paramspecial : "sret" | "link" | "fp" | "csr" | "vmctx" | "sigid" | "stack_limit"
-    callconv     : "fast" | "cold" | "system_v" | "fastcall" | "baldrdash_system_v" | "baldrdash_windows"
+```
+ signature    : "(" [paramlist] ")" ["->" retlist] [call_conv]
+ paramlist    : param { "," param }
+ retlist      : paramlist
+ param        : type [paramext] [paramspecial]
+ paramext     : "uext" | "sext"
+ paramspecial : "sret" | "link" | "fp" | "csr" | "vmctx" | "sigid" | "stack_limit"
+ callconv     : "fast" | "cold" | "system_v" | "fastcall" | "baldrdash_system_v" | "baldrdash_windows"
```
Comment (bjorn3) [inline cranelift/docs/ir.md]:
```suggestion
signature    : "(" [paramlist] ")" ["->" retlist] [call_conv]
paramlist    : param { "," param }
retlist      : paramlist
param        : type [paramext] [paramspecial]
paramext     : "uext" | "sext"
paramspecial : "sret" | "link" | "fp" | "csr" | "vmctx" | "sigid" | "stack_limit"
callconv     : "fast" | "cold" | "system_v" | "fastcall" | "baldrdash_system_v" | "baldrdash_windows"
```

File: cranelift/docs/heap.dot [hunk 6]
```diff
@@ -0,0 +1,8 @@
+digraph {
+        node [
+              shape=record,
+              fontsize=10,
+              fontname="Vera Sans, DejaVu Sans, Liberation Sans, Arial, Helvetica, sans"
+              ]
```
Comment (sunfishcode) [inline cranelift/docs/heap.dot]: As a super minor nit: this `]` should be indented at the level of the `node [`, for consistency with how bracketed constructs are typically formatted in Rust.

**Pull request merged**